### PR TITLE
Perf: Less views

### PR DESCRIPTION
**Pull request by chrisnojima**
I was looking at our view hierarchy as I was navving around and saw a bunch of useless views that could be removed. This PR does this.

For comparison:
```
branch:
    profile: 181 263
    folder: 146 175
    search: 70 76 
    chat: 720 784
    settings: 58 69

master:
    profile: 214 298
    folder: 162 192
    search: 89 96 
    chat: 734 799
    settings: 72 84

```

### CODE REVIEW

File: shared/common-adapters/hoc-timers.js [hunk 1]
```diff
@@ -17,8 +17,13 @@ function clearId(clearFunc: (id?: number) => void, array: Array<number>, id?: ?n
   }
 }
 
+function getDisplayName(WrappedComponent) {
```
Comment (chrisnojima) [inline shared/common-adapters/hoc-timers.js]: this passes through the displayname so we can tell what this is. was super confusing in the react view

File: shared/common-adapters/icon.native.js [hunk 2]
```diff
@@ -57,16 +57,17 @@ const Icon = (props: Exact<Props>) => {
...
-      activeOpacity={0.8}
-      underlayColor={props.underlayColor || globalColors.white}
-      onClick={props.onClick}
-      style={boxStyle}
-    >
-      {icon}
-    </ClickableBox>
-  )
+
+  return props.onClick
```
Comment (chrisnojima) [inline shared/common-adapters/icon.native.js]: not clickable? no wrapper

File: shared/global-errors/index.native.js [hunk 3]
```diff
@@ -87,6 +87,10 @@ class GlobalError extends Component<void, Props, State> {
   }
 
   render() {
+    if (this.state.size === 'Closed') {
```
Comment (chrisnojima) [inline shared/global-errors/index.native.js]: drawing nothing, return nothing

File: shared/main-shared.native.js [hunk 4]
```diff
@@ -72,16 +71,17 @@ class Main extends Component<void, any, void> {
 
     // TODO: move Push prompt into route
     const {showPushPrompt, mountPush} = this.props
+
+    if (showPushPrompt && mountPush) {
```
Comment (chrisnojima) [inline shared/main-shared.native.js]: this is really a different rendering path and not a child

File: shared/nav.native.js [hunk 5]
```diff
@@ -125,19 +130,18 @@ const forAndroid = ({hideNav, shim, tabBar}) => (
...
-        stack={screens}
-        renderRoute={renderStackRoute}
-        onNavigateBack={props.navigateUp}
-      />
-      {![chatTab].includes(props.routeSelected) &&
-        <Offline reachability={props.reachability} appFocused={true} />}
-      <GlobalError />
-    </Box>
-  )
+  const shim = [
```
Comment (chrisnojima) [inline shared/nav.native.js]: this is always in a box so just return an array. with react fiber this is the real fix for a lot of these issues cause you can return and array from a render and you don't have to always return a single item

**Comment by chrisnojima** [comment]
@keybase/react-hackers

**Comment by keybase-ci-visdiff** [comment]
The commits d771cba378f0...b20d75fbb425 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Tracker-New_user_Scroll_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498578761335-d771cba378f0-b20d75fbb425-linux/Tracker-New_user_Scroll_1-d771cba378f0.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498578761335-d771cba378f0-b20d75fbb425-linux/Tracker-New_user_Scroll_1-b20d75fbb425.png)  
   Tracker-New_user_Scroll_1 comparison
</summary>

**Pull request merged**